smarts/ros/__init__.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (C) 2021. Huawei Technologies Co., Ltd. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+
+import os
+
+# Note:  use of this module may require installing the SMARTS package with the "[ros]" extensions.
+
+# The following conditional import allows for extensions of the
+# ROSDriver node to import it using a much simpler path, like:
+#     from smarts.ros import ROSDriver
+# This can only be done when the code is being used in a ROS environment
+# (since ROSDriver requires the autogenerated message files),
+# hence the conditional import.
+_ros_pkg_path = os.environ.get("ROS_PACKAGE_PATH")
+if _ros_pkg_path and "smarts" in _ros_pkg_path:
+    from .src.smarts_ros.scripts.ros_driver import ROSDriver
```

Comment:
Yeah, as the comment right above it says, it only works when a user is in a ROS environment.  So I added the `if` to prevent it from failing if someone tries to, say, run the script in `__main__` or just do `import smarts.ros` somewhere.